composer.json

```diff
@@ -39,7 +39,9 @@
     "require": {
         "php": "^5.5.0|^7.0",
         "payum/iso4217": "~1.0",
-        "guzzlehttp/guzzle": "~6.0",
+        "php-http/client-implementation": "^1.0",
+        "php-http/discovery": "^0.8",
+        "guzzlehttp/psr7": "^1.1",
```

Comment:
Since we have required a virtual package (`php-http/client-implementation`) composer will fail if we are missing a package that provides `php-http/client-implementation`. The virtual package guarantees **an** implementation of a php-http client but not which one. You can consider a virtual package like an interface. 

Guzzle has two packages. One PSR7 implementation and one HttpClient. I have removed the dependency of the HttpClient. We are still dependent on the PSR7 implementation. I could do some work and remove the dependency of the PSR7 implementation as well.